sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -835,7 +849,8 @@ private void recordStoreResponseStatistics(
                 CosmosMeterOptions ruOptions = clientAccessor.getMeterOptions(
                     client,
                     CosmosMetricName.REQUEST_SUMMARY_DIRECT_REQUEST_CHARGE);
-                if (ruOptions.isEnabled()) {
+                if (ruOptions.isEnabled() &&
```

Comment:
maybe extract the condition checking into a method

Reply:
No - had that initially - reverted it because it simply is easier to read/understand this way.